doc/source/dev/contributor/debugging_linalg_issues.rst

```diff
@@ -0,0 +1,440 @@
+version, with ``NEWLAPACK`` symbols). If those limitations don't matter for
+you, FlexiBLAS can be a quite useful tool for efficient debugging, including
+for versions of OpenBlAS and other BLAS libraries that you have to build from
+source.
+
+Once you have everything set up, the development experience is::
+
+    $ python dev.py build -C-Dblas=flexiblas -C-Dlapack=flexiblas
+    $ FLEXIBLAS=NETLIB python dev.py test -s linalg
+    $ FLEXIBLAS=OpenBLAS python dev.py test -s linalg
+    # Or export the environment variable to make the selection stick:
+    $ export FLEXIBLAS=OpenBLAS
+
+You can also provide a path to a built BLAS library (e.g.,
+``FLEXIBLAS="libbhlas_atlas.so"``) - see the `usage docs in its README
+<https://github.com/mpimd-csc/flexiblas#selecting-the-backend-at-runtime>`__
+for more details.
+
+Unless you're on Fedora, you will likely have to build FlexiBLAS from source,
+which is a bit of work. The good news is that this should work no matter if
+you're on Linux or macOS, and use Python via virtualenvs, conda environments,
+or in some other way. We'll go through how to build OpenBLAS and FlexiBLAS
+from source, to allow debugging whether something in the latest OpenBLAS
+version is different from Netlib BLAS/LAPACK (or MKL) or not.
+
+The below should work in any environment where you can build SciPy itself; the
+only additional tool we need is CMake (install with, for example, ``pip install
+cmake``).
+
+Clone each repository::
+
+    $ cd ..  # starting from the root of the local `scipy` repo
+    $ mkdir flexiblas-setup && cd flexiblas-setup
+    $ git clone https://github.com/OpenMathLib/OpenBLAS.git openblas
+    $ git clone https://github.com/mpimd-csc/flexiblas.git
+    $ mkdir built-libs  # our local prefix to install everything to
+
+Build OpenBLAS::
+
+    $ cd openblas
+    $ mkdir build && cd build
+    $ cmake .. -DBUILD_SHARED_LIBS=ON -DCMAKE_INSTALL_PREFIX=$PWD/../../built-libs
+    $ cmake --build . -j
+    $ cmake --install . --prefix $PWD/../../built-libs
+    $ cd ../..
+
+Build FlexiBLAS::
+
+    $ mkdir build && cd build
+    $ # Note: this will also pick up the libraries in your system/env libdir
+    $ cmake .. -DEXTRA="OpenBLAS" -DLAPACK_API_VERSION=3.9.0 \
+        -DOpenBLAS_LIBRARY=$PWD/../../built-libs/lib/libopenblas.so \
+        -DCMAKE_INSTALL_PREFIX=$PWD/../../built-libs
+    $ cmake --build . -j
+    $ cmake --install . --prefix $PWD/../../built-libs
+    $ cd ../..
+
+We're now ready to build SciPy against FlexiBLAS::
+
+    $ cd scipy
+    $ python dev.py build -C-Dblas=flexiblas -C-Dlapack=flexiblas
+    ...
+    Run-time dependency flexiblas found: YES 3.4.2
+
+Now we can run the tests. Note that the ``NETLIB`` option is built without
+having to specify it; it's the default in FlexiBLAS and sources are included in
+its repository::
+
+    $ FLEXIBLAS=OpenBLAS python dev.py test -s linalg
+    $ FLEXIBLAS=NETLIB python dev.py test -s linalg
```

Comment:
Would be nice to comment what is the default, for `$ python dev.py test -s linalg` if FLEXIBLAS env var is not defined.

Reply:
done

Reply:
Why is netlib the default, not openblas?

Reply:
That's the choice made by FlexiBLAS, probably because Netlib is the reference and beyond that there's a host of accelerated libraries. It's authors choice; `numpy.distutils` preferred MKL over OpenBLAS when detected for example.